cmd/server/app/migrate_up.go

```diff
@@ -103,63 +104,31 @@ var upCmd = &cobra.Command{
 		cmd.Println("Database migration completed successfully")
 
 		cmd.Println("Ensuring authorization store...")
+		l := zerolog.Ctx(ctx)
 
-		authzw, err := authz.NewAuthzClient(&cfg.Authz)
+		authzw, err := authz.NewAuthzClient(&cfg.Authz, l)
 		if err != nil {
 			return fmt.Errorf("error while creating authz client: %w", err)
 		}
 
-		if !authzw.StoreIDProvided() {
-			storeID, err := ensureAuthzStore(ctx, cmd, authzw)
-			if err != nil {
-				return err
-			}
-
-			authzw.GetClient().SetStoreId(storeID)
+		if err := authzw.MigrateUp(ctx); err != nil {
+			return fmt.Errorf("error while running authz migrations: %w", err)
 		}
 
-		mID, err := authzw.WriteModel(ctx)
-		if err != nil {
-			return fmt.Errorf("error while writing authz model: %w", err)
-		}
-
-		if err := authzw.GetClient().SetAuthorizationModelId(mID); err != nil {
-			return fmt.Errorf("error setting authz model ID: %w", err)
+		if err := authzw.PrepareForRun(ctx); err != nil {
```

Comment:
Since `PrepareForRun` sets the `StoreId`, it need to be called before `MigrateUp`.

Reply:
`MigrateUp` makes sure the store and model exist. So `MigrateUp` actually needs to go first. Once we have a store and a model, then we can actually prepare the client for operations with OpenFGA.

Reply:
oh... this used to work because I set the StoreID while doing the migration before and now I don't. Let me fix this.

Reply:
Fixed